classes/author/maps/Schema.php

```diff
@@ -91,12 +93,9 @@ protected function mapByProperties(array $props, Author $item): array
         foreach ($props as $prop) {
             switch ($prop) {
                 case 'userGroupName':
-                    $userGroupId = $item->getData('userGroupId');
-
-                    $output[$prop] = $this->authorUserGroups->contains('userGroupId', $userGroupId)
-                        ? $this->authorUserGroups->firstWhere('userGroupId', $userGroupId)->getName(null)
-                        : '';
-
+                    /** @var UserGroup $userGroup */
+                    $userGroup = $this->authorUserGroups->first(fn (UserGroup $userGroup) => $userGroup->getId() === $item->getData('userGroupId'));
+                    $output[$prop] = $userGroup ? $userGroup->getName(null) : new stdClass();
```

Comment:
This addresses a `json_encode` issue with associate array vs. object. `$userGroup->getName(null)` returns an assoc array, which is converted to an object in JSON, so the "empty" equivalent has to be an object in PHP.

Since `Schema` maps are only ever used for the REST API, it makes sense to map them to values appropriate for `json_encode`. But if we ever use them separately, we might have to rethink this.